src/NuGet.Jobs.Catalog2Registration/Schema/EntityBuilder.cs

```diff
@@ -169,6 +169,15 @@ public void UpdateLeafItem(RegistrationLeafItem leafItem, HiveType hive, string
 
             catalogEntry.Title = packageDetails.Title ?? string.Empty;
             catalogEntry.Version = parsedVersion.ToFullString();
+
+            if (packageDetails.Vulnerabilities != null && packageDetails.Vulnerabilities.Count > 0)
+            {
+                catalogEntry.Vulnerabilities = packageDetails.Vulnerabilities;
+            }
+            else
+            {
+                catalogEntry.Vulnerabilities = new List<NuGet.Protocol.Catalog.PackageVulnerability> { new NuGet.Protocol.Catalog.PackageVulnerability() };
```

Comment:
Should we set this to `null` if there's no vulnerabilities? That seems to be the pattern we use if there's no deprecation.

Reply:
It is currently emitting a single empty vulnerability. I was simply following the lead of https://github.com/NuGet/NuGet.Services.Metadata/blob/b2a5100fb6111ddb569eb74af0cb2bc88918a22d/src/NuGet.Jobs.Catalog2Registration/Schema/EntityBuilder.cs#L167 for consistency. However, I expect that for actual processing of vulnerabilities client-side it would be neater to have nothing here. So, I could go either way on this, and generally opt for consistency in scenarios where I'm otherwise ambivalent. But if you have a strong opinion on this, I'm happy to be swayed. :)

Reply:
Hm the tags point is interesting. I'm not sure why we chose to have an empty string as the default, that may be vestigial from when we used JSON-LD and RDF to generate registration blobs.

> However, I expect that for actual processing of vulnerabilities client-side it would be neater to have nothing here.

I agree. Currently, clients will need to know how to interpret an empty vulnerability object. This does seem like a drawback.

> It is currently emitting a single empty vulnerability.

Keep in mind that unless we rebuild registration, we'll have a mix of registration blobs with no vulnerabilities and blobs with a single empty vulnerability. It seems like being consistent across all our blobs on the no vulnerabilities case would be better than being consistent with tags.

> But if you have a strong opinion on this, I'm happy to be swayed. :)

I'm currently leaning towards not emitting anything if a package has no vulnerabilities, but I could be swayed too 😄

Reply:
All good--you've sold me. I've made this change. Thanks!